HSDSolver/HSDlinsysolve.m

```diff
@@ -32,11 +32,11 @@
 const = 1e-2/max(1,norm(par.dy2));
 alpha = max(1e-14,min(1e-10,const*norm(par.rp))/(1+norm(diagschur.*par.dy2)));
 pertdiag = alpha*max(1e-8,diagschur); %% Note: alpha is close to 1e-15.
-mexschurfun(schur,pertdiag);
+    schur = mexschurfun(schur,pertdiag); 
```

Comment:
One of the key functional changes: eliminating the reliance on MEX side effects. This is but one of the many instances of this change.